server/src/db/association.rs

```diff
@@ -89,3 +113,103 @@ impl DatabaseAssociation {
         Ok(auth_iter.collect::<Result<Vec<_>, rusqlite::Error>>()?)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::test;
+    use shared::{CidrContents, Error};
+
+    use super::*;
+
+    #[tokio::test]
+    async fn test_double_add() -> Result<(), Error> {
+        let server = test::Server::new()?;
+
+        let contents = AssociationContents {
+            cidr_id_1: 1,
+            cidr_id_2: 2,
+        };
+        let contents_flipped = AssociationContents {
+            cidr_id_1: 2,
+            cidr_id_2: 1,
+        };
+        let res = server
+            .form_request(
+                test::ADMIN_PEER_IP,
+                "POST",
+                "/v1/admin/associations",
+                &contents,
+            )
+            .await;
+        assert!(res.status().is_success());
+
+        let res = server
+            .form_request(
+                test::ADMIN_PEER_IP,
+                "POST",
+                "/v1/admin/associations",
+                &contents,
+            )
+            .await;
+        assert!(res.status().is_client_error());
+
+        let res = server
+            .form_request(
+                test::ADMIN_PEER_IP,
+                "POST",
+                "/v1/admin/associations",
+                &contents_flipped,
+            )
+            .await;
+        assert!(res.status().is_client_error());
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn test_nonexistent_cidr_id() -> Result<(), Error> {
+        let server = test::Server::new()?;
+
+        // Verify both provided CIDR IDs exist
+        let last_cidr_id: i64 =
+            server
+                .db()
+                .lock()
+                .query_row("SELECT COUNT(*) FROM cidrs", params![], |r| r.get(0))?;
+        let contents = AssociationContents {
+            cidr_id_1: 1,
+            cidr_id_2: last_cidr_id + 1,
+        };
+        let res = server
+            .form_request(
+                test::ADMIN_PEER_IP,
+                "POST",
+                "/v1/admin/associations",
+                &contents,
+            )
+            .await;
+        assert!(!res.status().is_success());
```

Comment:
Super-nit: using `.is_client_error()` as already done above could be a tiny bit more descriptive here.